core/AuthResult.php

```diff
@@ -16,6 +16,8 @@
  */
 class AuthResult
 {
+    const FAILURE_POSTONLY_TOKEN = -2;
+    const FAILURE_EXPIRED_TOKEN = -1;
```

Comment:
I was working from this sentence in the issue specification: 

>When someone tries to create a token that doesn't allow usage in URL parameter and tries to use it in a url parameter, we should have a proper error message.

Much easier not to do this and just keep the same behaviour, I'll ignore this part of the spec and rework this PR not to provide specific error messages :+1:

Reply:
That would be still good to have. Sorry I thought it was around expired token. We would show the same message though to valid and invalid tokens. In that case it wouldn't be a security issue.

Reply:
Ok, I'll remove the message for the expired token failure and just keep the error message for the post only failure.

Reply:
Personally I wouldn't check if the token, that was sent, would have been required to be sent as post in order to be able to print another error. I would suggest to simply have two different message in case a token_auth was the sent with the request, but couldn't be validated.
If the token was sent via GET the message could be something like: `Unable to authenticate with the provided token. It is either invalid, expired or is required to be sent as POST parameter`.
If the token was sent via POST parameter it would only say `Unable to authenticate with the provided token. It is either invalid or expired.`
That would remove a lot complexity, but still give the user a hint, where the problem could be.
@tsteur would that be enough that way?

Reply:
That would be enough that way 👍

Reply:
There are now just two error messages shown as above, depending on whether the token was posted or not. The additional `AuthResult` values have been removed.